### DISCUSSION TITLE
Set Default Collection Fields for API Responses

### DISCUSSION THREAD

**Discussion by ChappIO** [directus/directus#6623] [Feature Requests]
I am currently working on a heavily relational dataset with over 14.2 million records on the root collection. There are several one-to-many relationships on this root collection. As you can imagine, the REST api calls to the listing take quite a while.

I would love a way to disable/hide a field on the collection (/items/<collection_name>) endpoint so people cannot overload my server with listing requests when they are too lazy to specify the required fields.

P.S. Amazing product, I'm working with the release candidate right now and it has been rock solid.

**Comment by benhaynes** [comment]
Thank you, and I like this idea. Some thoughts:

* You obviously know that you can limit the fields in the response using the [`?fields` parameter](https://docs.directus.io/reference/api/query/#fields).
* To avoid malicious usage, you could [limit permissions](https://docs.directus.io/guides/permissions/) for `public` or your dedicated API `role`.

But regarding your request, we like the idea of setting the default response fields (per Collection). Basically, we would add a "Default Response Fields" option under Collection settings, this _tree view interface_ would let you configure what field data is returned when _not_ using the `?fields` param. Not only would this let you remove some parent fields, it would also allow you to include deep relational data by default.

Thoughts on the above?

**Reply by ChappIO** [reply]
Yes that would be great! And yes I also know you can specify the fields property but my experience tells me most people do not. I intend to expose this api to many clients so being able to specify the default fields for the "summary" view would be ideal.

**Reply by rijkvanzanten** [reply]
Just to confirm, you still want to allow this user group to fetch _more_ data when needed right? Or are you trying to limit the available data altogether?

**Reply by ChappIO** [reply]
Correct, I want to allow them to pull whatever they want but I don't want to spend needless cycles on lazy devs that don't specify a field list.

**Reply by philleepflorence** [reply]
Concur with the idea, use case and thoughts...

My current use case:
- manage `SELECT * FROM...` people (sorta like default display settings),
- hide `presentation` or `metadata` fields (sorta like hide from items and item settings)

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
